keras/utils/conv_utils.py

```diff
@@ -49,6 +49,31 @@ def convert_data_format(data_format, ndim):
         'Expected values are ["channels_first", "channels_last"]')
 
 
+def normalize_positive_tuple(value, n, name):
+  """Transforms a single positive integer or iterable of positive integers into an
+    integer tuple.
+
+  Args:
+    value: The value to validate and convert. Could a positive int, or any iterable
+      of positive ints.
+    n: The size of the tuple to be returned.
+    name: The name of the argument being validated, e.g. "pool_size" or "kernel_size".
+    This is only used for format error messages.
+
+  Returns:
+    A tuple of n positive integers.
+
+  Raises:
+    ValueError: If something else than an postive int/long or iterable thereof was passed.
+  """
+  value_tuple = normalize_tuple(value, n, name)
+  for value in value_tuple:
+    if value <= 0:
```

Comment:
Please find all the negative values, rather than just the first one.

Reply:
For strides, someone may want to slide a window with the same width to the input width along the length dimension. There is a possibility that the stride is zero at a dimension while not at other dimensions.

Reply:
So I try a more flexible way as discussed above.